### PR TITLE
PR1 fixes / improvements

### PR DESCRIPTION
**Pull request by ivanistheone**
- use YouTubeVideoFile instead of VideoFile  (just need to specify the `youtube_id` and ricecooker will take care of downloading the video, and storing it in `storage/` directly
  - set the youtube_dl  flag to download=False (chef just needs to get the json metadata; the video download happens later)
  - made changes to ensure `content_id = f(source_id)` stay constant between runs --- if `source_id` changes randomly on each chef run this channel cannot be remixed 
  - implemented youtube_dl json data caching. Run the chef with the --update flag to clear youtube json info cache

### CODE REVIEW

File: arvind.py [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+YOUTUBE_CACHE_DIR = os.path.join('chefdata', 'youtubecache')
+YOUTUBE_ID_REGEX = re.compile(r'(https?://)?(www\.)?(youtube|youtu|youtube-nocookie)\.(com|be)/(watch\?v=|embed/|v/|.+\?v=)?(?P<youtube_id>[A-Za-z0-9\-=_]{11})')
+
 # List of languages not avialble at the le_utils
 UND_LANG = {
             "marwari":{
                 "name":"Marwari",
                 "native_name":"marwari",
-                "code": "mwr",
+                "code": "und",      # temporary while le-utils updated in Studio
```
Comment (ivanistheone) [inline arvind.py:18]: The `mrw` and `bho` language codes are not deployed yet in Studio, so for the time being use `und`

File: arvind.py [hunk 2]
```diff
@@ -124,32 +131,49 @@ def download(self, download_dir="./"):
...
+            vinfo = json.load(open(vinfo_json_path))
+
+        # else get using youtube_dl:
+        else:
+            with youtube_dl.YoutubeDL(ydl_options) as ydl:
+                pp = pprint.PrettyPrinter()
+                try:
+                    ydl.add_default_info_extractors()
+                    # vinfo = ydl.extract_info(self.url, download=True)
+                    vinfo = ydl.extract_info(self.url, download=download)
```
Comment (ivanistheone) [inline arvind.py:153]: Here, `download` is now set to `False` so the call `ydl.extract_info` will only get the video info json data and not download the actual file. Runs very fast...

**Comment by ivanistheone** [comment]
@mrpau-richard @cpauya  Please review and merge these fixes into your PR and re-run the chef script. I tested only with a sample so not sure if it is 100%. 

We can then merge your https://github.com/learningequality/sushi-chef-arvind-gupta-toys/pull/1 on the main repo.

**Comment by mrpau-richard** [comment]
Thanks for this @ivanistheone, there are errors on downloading and creating youtubecache `.json` and some variables not use in this PR is not removed, let me fix it. merging this.

**Pull request merged**